### ISSUE TITLE
Feature Request: Configure icon-pin placement

### ISSUE THREAD

**Issue by lexicalunit** [paulpflug/opened-files#6]
I think it looks a bit nicer to the left of the file name. Usually my tree-view isn't sized large enough to even see the pin icons if they're placed to the right of the file name, so I was mostly unable to take advantage of that feature during normal use.

![screen shot 2015-06-23 at 3 41 21 pm](https://cloud.githubusercontent.com/assets/1903876/8316993/096252fe-19c0-11e5-8d13-5bfdf9eddba7.png)

I tried to adjust things using just my personal `styles.less` file but was unable to make it work correctly. I tried using `float: left` but that caused a newline to be inserted between the icon and the name when the view was sized to be too small to fit the entire file name. Then I tried using `direction: rtl` but that caused overflow to scroll off the left edge, obscuring the pin icon.

Finally I resorted to modifying `file.vue` and when that didn't work, modifying the `file.js` variable `__vue_template__`. I assume there's some way to compile the `vue` file into a `js` file based on the naming of the directory, but it wasn't obvious to me what CLI tool I would use to do that.

Here's my fork: https://github.com/lexicalunit/opened-files/tree/lexicalunit_tweak. I didn't create a PR because I assume you'll want to implement this as a configuration setting.

I also used my `styles.less` file to:
- Remove the icon-x after file names.
- Adjust the margin between the directory names and icon-x.
- Disabled transition effects.
- Disabled box-shadow effects.
- Generally made the view look more like the tree-view in small ways.

I only mention this so it's more clear what you're seeing in my screenshot. Here's the `less` code:

``` less
.of-highlighted, ul.tab-bar > li.tab.of-highlighted {
    background-color: inherit;
    transition-property: none;
    transition-duration: 0;
}

.of-unhighlighted, ul.tab-bar > li.tab.of-unhighlighted {
    background-color: inherit;
    transition-property: none;
    transition-duration: 0;
}

.file-list.opened-files {
  box-shadow: none;
  border-bottom: 1px solid @syntax-wrap-guide-color;
  border-right: 1px solid @syntax-wrap-guide-color;
}

.file-list .list-item.file {
    .icon-x {
        display: none;
    }
}

.file-list .directory {
    .icon-x {
        margin-left: .5em;
    }
}
```

**Comment by paulpflug** [comment]
Wow, that looks nice. I will definitely include it.
Secretly I hoped somebody will do the styling for me 😁
Thanks

Yes there is a vue - js compiler.
You have to call 'npm install' in the package directory and open the dir in 
dev mode. Then it will compile and reload on file save.
I really should include that info in the Readme 😁

Am 23. Juni 2015 22:56:31 schrieb lexicalunit notifications@github.com:

**Comment by paulpflug** [comment]
I have reworked appearance and behavior (inspired by VS code and your changes)

Sadly I can't properly work with the preview tabs, yet :(
(preview tabs shouldn't be added to the list, but I can't catch it)
All tabs are pinned now by default and unpinning is replaced by unpinning+close
Makes sense in my eyes..

I also added an option to disable hover coloring (but I also dimmed it)
The Icon in the front is way nicer. I added hide on not hover.
I have additionally removed the folder icon.

What do you think?

**Comment by lexicalunit** [comment]
Hrm. So opened-files works on my work laptop, but it doesn't seem to work on my home laptop. It's installed and enabled, and it's commands show up in my command palette, but the view just isn't there. Trying to diagnose on my end...

**Comment by lexicalunit** [comment]
I'm also not able to get the logging or vue compiler to work. I've cloned the package into my `~/.atom/packages` directory and opened the project in Atom's dev mode, but I get no logging, and changing the vue files doesn't seem to kick off anything.

**Comment by lexicalunit** [comment]
So I've traced the code that gets executed in `main.coffee` when I force a window reload. We eventually get into the branch of code here:

``` coffee
        if @compiling?
          @compiling.then load
```

But the `load` callback is never actually entered...

**Comment by lexicalunit** [comment]
Got it! It's because my home laptop doesn't have a `~/.atom/projects.cson` file. This causes something within

``` coffee
            OpenedFiles ?= require "./#{pkgName}"
            @openedFiles = new OpenedFiles(logger)
```

to throw an exception saying that the file does not exist. I googled for this file but didn't see anything really concrete about it. If you're expecting this file to exist, you might need to create it yourself if it's missing.

**Edit: This doesn't resolve the issue in dev mode where the then clause of `@compiling.then load` never executes. I am able to get opened-files view to show in non-dev mode only.**

**Comment by lexicalunit** [comment]
With `As List` enabled and `Highlight On Hover` **disabled** this is what I get when I move my mouse over on the the files in the list. They just sort of disappear, leaving a blank spot. Here I have my mouse over `folder.js`:

![screen shot 2015-06-24 at 9 49 28 pm](https://cloud.githubusercontent.com/assets/1903876/8345902/03a59b90-1abb-11e5-8851-31b3e1d4d3c0.png)

When `As List` is disabled the `Highlight On Hover` doesn't seem to do anything, but when I hover over an item the text lightens up a bit, but the background color remains unchanged.

**Comment by paulpflug** [comment]
could you try it again with 0.2.1? 
I couldn't reproduce the hovering issue.

I will setup a clean atom env. to test the dev-mode issue

**Comment by paulpflug** [comment]
closed due to age - there is a new version 0.3.0
